File: desh/ege2026kp/25solve/25-178.pas
## uses school;
(500001..maxInt).Select( x->x.Divisors )
  .Select( divs->(divs[^1], 
         divs.Where( d->(d mod 10 = 8) and
                        (d in 18..(divs[^1]-1))).ToArray ) )
  .Where( \(x,d)->d.Count > 0  )
  .Select( \(x,d)-> x + ' ' + d[0] )
  .Take(5)
  .PrintLines;

File: desh/ege2026kp/25solve/25-178.py
start = 500000

count = 0
x = start
while count < 5:
  d = 18
  while d < x:
    if x % d == 0:
      break
    d += 10
  if d < x:
    print( x, d )
    count += 1
  x += 1